src/main/java/uk/gov/hmcts/dts/fact/services/admin/list/AdminOpeningTypeService.java

```diff
@@ -0,0 +1,141 @@
+
+    @Autowired
+    public AdminOpeningTypeService(final OpeningTimeRepository openingTimeRepository,
+                                   final OpeningTypeRepository openingTypeRepository,
+                                   final AdminAuditService adminAuditService) {
+        this.openingTimeRepository = openingTimeRepository;
+        this.openingTypeRepository = openingTypeRepository;
+        this.adminAuditService = adminAuditService;
+    }
+
+    public List<OpeningType> getAllOpeningTypes() {
+        return openingTypeRepository.findAll()
+            .stream()
+            .map(OpeningType::new)
+            .sorted(Comparator.comparing(OpeningType::getType))
+            .collect(toList());
+    }
+
+
+    public OpeningType getOpeningType(final Integer id) {
+        try {
+            return new OpeningType(openingTypeRepository.getOne(id));
+        } catch (final javax.persistence.EntityNotFoundException exception) {
+            throw new NotFoundException(exception);
+        }
+    }
+
+    @Transactional
+    public OpeningType createOpeningType(final OpeningType openingType) {
+        checkIfOpeningTypeAlreadyExists(openingType.getType());
+        OpeningType newOpeningType  = new OpeningType(openingTypeRepository.save(createNewOpeningTypeEntityFromModel(openingType)));
+
+        adminAuditService.saveAudit(
+            AuditType.findByName("Create opening type"),
+            openingType,
+            newOpeningType,
+            null);
+        return newOpeningType;
+    }
+
+
+    @Transactional
+    public OpeningType updateOpeningType(final OpeningType updatedOpeningType) {
+        uk.gov.hmcts.dts.fact.entity.OpeningType openingTypeEntity =
+            openingTypeRepository.findById(updatedOpeningType.getId())
+                .orElseThrow(() -> new NotFoundException(updatedOpeningType.getId().toString()));
+
+        checkIfUpdatedOpeningTypeAlreadyExists(updatedOpeningType);
+        openingTypeEntity.setDescription(updatedOpeningType.getType());
+        openingTypeEntity.setDescriptionCy(updatedOpeningType.getTypeCy());
+
+        OpeningType newOpeningType  =  new OpeningType(openingTypeRepository.save(openingTypeEntity));
+
+        adminAuditService.saveAudit(
+            AuditType.findByName("Update opening type"),
+            getOpeningType(updatedOpeningType.getId()),
+            newOpeningType,
+            null);
+        return newOpeningType;
+
+
+    }
+
+    public void deleteOpeningType(final Integer openingTypeId) {
+        checkOpeningTypeIsNotInUse(openingTypeId);
+
+        adminAuditService.saveAudit(
+            AuditType.findByName("Delete opening type"),
+            getOpeningType(openingTypeId),
+            null,
+            null);
+
+        try {
+            openingTypeRepository.deleteById(openingTypeId);
+        } catch (EmptyResultDataAccessException ex) {
+            log.warn("Opening Type could not be deleted because it no longer exists: " + openingTypeId);
```

Comment:
Up to you if you want to do formatting on this, but you can use the "% longer exists: {}", openingTypeId) structure instead 

For below, you can use String.format if you want, but yeah up to you

Reply:
It's fine the way it is to me.